controller/service.go

```diff
@@ -115,7 +115,7 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 				c.client.Errorf(svc, "LoadBalancerFailed", "Failed loadbalancer IP for dual-stack")
 				return true
 			}
-			if svc.Spec.LoadBalancerIP != lbIPs[0].String() {
+			if len(lbIPs) != 0 && svc.Spec.LoadBalancerIP != lbIPs[0].String() {
```

Comment:
I'd be on the paranoid side and clear also if `len(lbIPs) == 0 && svc.Spec.LoadBalancerIP != ""`

Reply:
len(lbIPs) == 0 at this point could only happen if the state was already cleared, should I still add it?

Reply:
Either in that scenario, or if the service comes in without ips assigned. Again, I would say that it's not necessary, but I prefer being on the paranoid side.

Reply:
Nevermind, I just noticed we are inside a `if len(lbIPs) != 0 {`